manim/utils/updaters.py

```diff
@@ -0,0 +1,218 @@
+        parameters = [str(param) for param in signature.parameters.values()]
+
+        for i, param in enumerate(parameters):
+            # Stop when finding **kwargs or parameters with default values
+            if param.startswith("**") or "=" in param:
+                num_non_default_parameters = i
+                break
+        else:
+            num_non_default_parameters = len(parameters)
+
+        if num_non_default_parameters == 0:
+            self._raise_error(0)
+
+        # If this is a method being called from an instance, exclude the 1st
+        # parameter if it's called "self"
+        if inspect.ismethod(updater) and parameters[0] == "self":
+            num_non_default_parameters -= 1
+        # Exclude the "cls" parameter from class methods
+        if parameters[0] == "cls":
+            num_non_default_parameters -= 1
+
+        # Handle functions containing *args, assuming that all can be passed
+        # a 2nd parameter dt
+        if 1 <= num_non_default_parameters <= 3 and parameters[-1].startswith("*"):
+            num_non_default_parameters = 2
+
+        if num_non_default_parameters == 1:
+            self.is_time_based = False
+        elif num_non_default_parameters == 2:
+            self.is_time_based = True
+        else:
+            self._raise_error(num_non_default_parameters)
+
+    def _raise_error(self, num_non_default_parameters: int):
+        updater_name = self.updater.__qualname__
+        signature = str(inspect.signature(self.updater))
+        full_name = updater_name + signature
+
+        if num_non_default_parameters == 0:
+            num_non_default_parameters = "no"
+
+        raise ValueError(
+            "An updater function must accept either 1 or 2 parameters without "
+            "default values (not including 'self' or 'cls' for methods), but "
+            f"the function {full_name} has {num_non_default_parameters} such "
+            "parameters."
+        )
+
+
+class MobjectUpdaterWrapper(AbstractUpdaterWrapper):
+    """Wraps a :class:`MobjectUpdater` function, inspects its signature and
+    calculates whether it's time-based or not.
+
+    If it has a single parameter (with no default value), it's considered
+    non-time-based: it doesn't depend on time.
+
+    If it has two parameters (with no default values), it's considered
+    time-based: it depends on time, and the affected Mobject has a change
+    on every frame which depends on the frame's duration dt.
+
+    .. note::
+        It's not mandatory that the parameters are named ``mob`` and ``dt``.
+
+    **Only parameters with no default values are considered in when determining
+    whether the updater is time-based or not.** For example, an updater
+    ``lambda mob, rate=5: ...`` is considered non-time-based since the 2nd
+    parameter ``rate`` has a default value of 5. **This allows for passing
+    functions with more than 2 parameters, as long as the extra parameters have
+    default values.**
```

Comment:
> This allows for passing functions with more than 2 parameters, as long as the extra parameters have default values.

Maybe don't bold this? Might be a good idea to get a third opinion on this :thinking: